Source/WebCore/page/EventCounts.h

```diff
@@ -26,21 +26,35 @@
 
 #pragma once
 
+#include "EventNames.h"
+#include "IDLTypes.h"
+#include "JSDOMMapLike.h"
 #include <wtf/RefCounted.h>
 
+namespace JSC {
+class JSGlobalObject;
+class JSObject;
+}
+
 namespace WebCore {
 
 class DOMMapAdapter;
 
 class EventCounts final: public RefCounted<EventCounts> {
 public:
-
-    static Ref<EventCounts> create() { return adoptRef(*new EventCounts()); }
-
+    static Ref<EventCounts> create() { return adoptRef(*new EventCounts); }
     void initializeMapLike(DOMMapAdapter&);
+    void add(EventType);
 
+    ScriptExecutionContext* scriptExecutionContext() const;
 private:
     EventCounts();
+    std::array<unsigned, EventNames::TimedEvents.size()> m_counts { };
+
+    // FIXME: the maplike IDL interface should read directly
+    // from m_counts instead of doing this
+    JSC::JSGlobalObject* m_maplikeGlobalObject;
```

Comment:
I don't think this is safe or correct. We can't store a raw pointer to JSGlobalObject/JSObject like this.